app/src/main/java/com/mobile/tandil/kotlinbaseproject/activity/MainActivity.kt

```diff
@@ -0,0 +1,48 @@
+package com.mobile.tandil.kotlinbaseproject.activity
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.mobile.tandil.kotlinbaseproject.R
+import com.mobile.tandil.kotlinbaseproject.databinding.ActivityMainBinding
+import com.mobile.tandil.kotlinbaseproject.fragment.MainDialogFragment
+import com.mobile.tandil.kotlinbaseproject.listener.ListenerSetParkingSpaces
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.MainViewModel
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.MainViewModel.CheckState
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.MainViewModel.ParkingData
+
+class MainActivity : AppCompatActivity(), ListenerSetParkingSpaces {
+
+    private lateinit var binding: ActivityMainBinding
+    private val viewModel = MainViewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.getValue().observe({ lifecycle }, ::updateUI)
+
+        setListeners()
+        viewModel.initialState()
+    }
+
+    private fun setListeners() {
+        binding.updateButton.setOnClickListener {
```

Comment:
this should call the viewModel, the viewModel set a new LiveData state and the activity react to that state

Reply:
done